src/acl/external/time_quota/Makefile.am

```
@@ -14,7 +14,15 @@ DEFS += -DDEFAULT_QUOTA_DB=\"$(localstatedir)/ext_time_quota.db\"
 
 ext_time_quota_acl_SOURCES = \
 	ext_time_quota_acl.cc
+
 ext_time_quota_acl_LDADD = \
+	$(top_builddir)/src/sbuf/libsbuf.la \
+	$(top_builddir)/src/debug/libdebug.la \
+	$(top_builddir)/src/error/liberror.la \
+	$(top_builddir)/src/comm/libminimal.la \
+	$(top_builddir)/src/mem/libminimal.la \
+	$(top_builddir)/src/base/libbase.la \
+	$(top_builddir)/src/time/libtime.la \
```

The dependencies can be slightly tricky to work out and we have a lot of technical debt in the existing lists.

For new additions; please use the `SUBDIRS` list in `src/Makefile.am` as the authoritative dependency sequence. `LDADD` should then list the needful objects in reverse order.

```suggestion
	$(top_builddir)/src/sbuf/libsbuf.la \
	$(top_builddir)/src/error/liberror.la \
	$(top_builddir)/src/comm/libminimal.la \
	$(top_builddir)/src/base/libbase.la \
	$(top_builddir)/src/debug/libdebug.la \
	$(top_builddir)/src/time/libtime.la \
	$(top_builddir)/src/mem/libminimal.la \
```

---

> For new additions; please use the `SUBDIRS` list in `src/Makefile.am` as the authoritative dependency sequence. `LDADD` should then list the needful objects in reverse order.
 
Let's not declare SUBDIRS order authoritative for LDADD: SUBDIRS determine _build_ dependency order among subdirectories. LDADD determines _linking_ order. One does not have to be the reversed version of the other (and LDADD order is often more strict/important/difficult to get right). In cases where one subdirectory contains multiple compatible (i.e. not mutually exclusive) libraries, SUBDIRS order cannot determine LDADD order at all!

For consistency sake, let's start new (or substantially new) LDADD lists with (reversed) SUBDIRS order and then adjust LDADD (and sometimes SUBDIRS!) as needed, without declaring SUBDIRS order as authoritative.

P.S. This comment is not endorsing or objecting to specific order suggested on this change request thread.

---

It is not clear to me whether this change should go in or not. The code as in this PR builds. Why should I change it? Why should I not change it? Why should fixing LDADD or SUDIRS beyond what is needed to build this PR be in scope for this PR?

---

I think I see the logic in the change. I agree to not setting additional rules - we have too many already. It already works but won't hurt so ahoy!

---

> It already works but won't hurt so ahoy!

It did hurt [AFAICT](https://github.com/squid-cache/squid/actions/runs/10323802951/job/28581955676?pr=1847). To avoid misunderstanding, I am still OK with any order you two (and CI) can agree on here.

---

reverted to previous order, which works